driver/src/main.rs

```diff
@@ -95,9 +106,16 @@ fn main() {
     info!("using options: {:#?}", options);
 
     let web3 = web3_provider(options.node_url.as_str(), options.rpc_timeout).unwrap();
-    let contract =
-        BatchExchange::new(&web3, options.private_key.clone(), options.network_id).unwrap();
-    info!("Using contract at {:?}", contract.address());
+    let gas_station =
+        GnosisSafeGasStation::new(options.gas_station_timeout, gas_station::DEFAULT_URI).unwrap();
+    let contract = StableXContractImpl::new(
+        &web3,
+        options.private_key.clone(),
+        options.network_id,
+        &gas_station,
+    )
+    .unwrap();
+    info!("Using contract at {:?}", contract.instance().address());
```

Comment:
ubernit: I'm generally not a fan of getter chains (as they leak information and arguably violate [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter#In_object-oriented_programming)).

Reply:
I changed it but do want to point out that we already had access to the whole `BatchExchange` before, just not through a getter, so that hasn't *changed*. At the same time it can still be better to hide it.

Reply:
yes good point. If there is a good reason to re-expose it I'm not against that.